src/transform/tests/testdata/predicate_propagation

```diff
@@ -287,14 +283,16 @@ build apply=PredicateKnowledge
 (filter (filter (get x) [(call_unary not (call_binary gt #0 #1))]) [(call_binary gt #0 #1)])
 ----
 %0 =
-| Constant
+| Get x (u0)
```

Comment:
This is a side-effect of calling `reduce` on the predicates during normalization. A more general range-analysis solution is needed anyway to cover all/more cases.